portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -114,12 +116,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
         UserService userService = UserServiceFactory.getUserService();
         String userEmail = userService.getCurrentUser().getEmail();
 
+        UUID id = UUID.randomUUID();
```

Comment:
This may be too paranoid in practice, but it may be good to generate the UUID inside a loop and query the datastore to check for a collision with a previous comment. I realize the probability is ridiculously low, but doing it properly makes us not depend on the implementation since the API doesn't guarantee no collisions.

pseudocode of what I mean:
```
id = new uuid
while id collides:
  id = new uuid
```

Reply:
I just added collision checking function.